src/tyro/_arguments.py

```diff
@@ -611,4 +611,4 @@ def _rule_apply_argconf(
     if arg.field.argconf.metavar is not None:
         lowered.metavar = arg.field.argconf.metavar
     if arg.field.argconf.aliases is not None:
-        lowered.name_or_flags = lowered.name_or_flags + arg.field.argconf.aliases
+        lowered.name_or_flags = arg.field.argconf.aliases + lowered.name_or_flags
```

Comment:
Ah. I guess that since all of the aliases are part of the same list, it would require extra work to pull just the short ones out and place them before everything else. In the issue that I filed, I wasn't looking to flip the order of the destination variable-derived name (`lowered.name_or_flags`) versus all of the aliases... only the short options (e.g., `-d`.).

However, I'm not against this change. Don't know if it might cause too much surprise to other Tyro users.

Reply:
Yeah, I think all of the options seemed similar:
1. Pull short args only to the front
2. Sort by length
3. Just order as `(*user args, automatic arg)`

But the final one just seemed the least magical

Reply:
That makes sense: keep it simple. Hopefully none of the other users will complain about the change in order.